### PR TITLE
=str - Minor improvements to SynchronousFilePublisher

### PR DESCRIPTION
**Pull request by viktorklang**

### CODE REVIEW

File: akka-stream/src/main/scala/akka/stream/impl/io/SynchronousFilePublisher.scala [hunk 1]
```diff
@@ -65,47 +65,43 @@ private[akka] class SynchronousFilePublisher(f: File, bytesReadPromise: Promise[
...
         eofReachedAtOffset = chan.position
         log.debug("No more bytes available to read (got `-1` from `read`), marking final bytes of file @ " + eofReachedAtOffset)
-
-      case _ ⇒
+      case 0 ⇒ // had nothing to read into this chunk
+      case readBytes ⇒
+        buf.flip()
         readBytesTotal += readBytes
-        availableChunks :+= ByteString(buf.array).take(readBytes)
+        availableChunks :+= ByteString.fromByteBuffer(buf)
```
Comment (ktoso) [inline akka-stream/src/main/scala/akka/stream/impl/io/SynchronousFilePublisher.scala]: Ah, nice one!

File: akka-stream/src/main/scala/akka/stream/impl/io/SynchronousFilePublisher.scala [hunk 2]
```diff
@@ -30,14 +31,13 @@ private[akka] object SynchronousFilePublisher {
...
-  extends akka.stream.actor.ActorPublisher[ByteString]
-  with ActorLogging {
-
+private[akka] final class SynchronousFilePublisher(f: File, bytesReadPromise: Promise[Long], chunkSize: Int, initialBuffer: Int, maxBuffer: Int)
+  extends akka.stream.actor.ActorPublisher[ByteString] with ActorLogging {
   import SynchronousFilePublisher._
 
   var eofReachedAtOffset = Long.MinValue
 
+  val buf = ByteBuffer.allocate(chunkSize)
```
Comment (ktoso) [inline akka-stream/src/main/scala/akka/stream/impl/io/SynchronousFilePublisher.scala]: Right, good idea.

**Comment by drewhk** [comment]
LGTM, nice!

**Comment by ktoso** [comment]
LGTM, thanks @viktorklang!

**Comment by viktorklang** [comment]
Thanks everyone, I updated it a bit more to reduce the number of var-manipulations

**Comment by ktoso** [comment]
Build failures unrelated, and also fixed now thanks to merging: https://github.com/akka/akka/pull/18642 .
I'll merge once PR validation completes here.

Thanks Viktor!

**Pull request merged**